### PR TITLE
[INTEL MKL] Compiler Bug Fix :  Typing problem in TF 2.2.0 With Visual Studio 2019

### PR DESCRIPTION
**Pull request by khaled-besrour**
https://github.com/tensorflow/tensorflow/issues/37430

### CODE REVIEW

**Comment by khaled-besrour** [comment]
@googlebot I signed it!

File: tensorflow/core/util/mkl_util.h
Comment (penpornk) [inline tensorflow/core/util/mkl_util.h]: Could you please show the error message related to this fix? I can't find it in https://github.com/tensorflow/tensorflow/issues/37430.

**Review by penpornk — COMMENTED** [review]
Thank you for the PR! I've looped in the @TensorFlow-MKL team to help review the PR.

File: tensorflow/core/util/mkl_util.h
Comment (khaled-besrour) [reply tensorflow/core/util/mkl_util.h]: the file moved a little, the index aren't the same, but the log for the error is here .
line equivalence 1262 = 1252

```
ERROR: C:/tensorflow/tensorflow/core/kernels/BUILD:8128:1: C++ compilation of rule '//tensorflow/core/kernels:mkl_aggregate_ops' failed (Exit 2)
.\tensorflow/core/util/mkl_util.h(1253): error C2131: expression did not evaluate to a constant
.\tensorflow/core/util/mkl_util.h(1252): note: failure was caused by a read of a variable outside its lifetime
.\tensorflow/core/util/mkl_util.h(1252): note: see usage of 'dim'
.\tensorflow/core/util/mkl_util.h(1254): error C2131: expression did not evaluate to a constant
.\tensorflow/core/util/mkl_util.h(1252): note: failure was caused by a read of a variable outside its lifetime
.\tensorflow/core/util/mkl_util.h(1252): note: see usage of 'dim'
.\tensorflow/core/util/mkl_util.h(1256): error C3863: array type 'dnnl_dim_t [kNumDims]' is not assignable
.\tensorflow/core/util/mkl_util.h(1257): error C3863: array type 'dnnl_dim_t [kNumDims]' is not assignable
Target //tensorflow/tools/pip_package:build_pip_package failed to build
```

Comment (penpornk) [reply tensorflow/core/util/mkl_util.h]: I see. Thank you!

**Review by penpornk — COMMENTED** [review]
The changes look good to me. I'll wait for the green light from the Intel TensorFlow-MKL team before I approve the PR. Thank you again! :)

**Comment by gbaned** [comment]
@TensorFlow-MKL  Any update on this PR? Please. Thanks!

**Review by penpornk — APPROVED** [review]
@TensorFlow-MKL just gave a green light to me offline. Approving the PR now. Thank you for your patience!

**Pull request merged**
